### DISCUSSION TITLE
Miniscript support in output descriptors

### DISCUSSION THREAD

**Discussion by darosior** [btcpayserver/btcpayserver#5091] [Ideas / Feature Requests]
Is there any plan to start supporting Miniscript inside output descriptors? Right now if i want to receive onchain payments to my wallet that makes use of Miniscript descriptors (Bitcoin Core, [Liana](https://github.com/wizardsardine/liana), ..) i need to create a temporary BTCPay-specific wallet to receive funds from my shop there before sending them again to my main wallet.

**Comment by Kukks** [comment]
Do you have any specific descriptors we can use to develop this feature against?

**Comment by darosior** [comment]
Are you asking about example of Miniscript descriptors? Or the specific instance i'm using?

**Comment by NicolasDorier** [comment]
@Kukks he doesn't want specific descriptor, he wants all miniscript descriptor.
That's a big job, as I need to port miniscript to NBitcoin.
I also need to modify NBXplorer to accept any output descriptor.
Then finally, we need to modify BTCPay to accept every output descriptor. (maybe nuking the "enter your xpub" option)
